test/joystick_test.py

```diff
@@ -31,21 +37,38 @@ def todo_test_Joystick(self):
 
 
 class JoystickModuleTest(unittest.TestCase):
-    def todo_test_get_count(self):
-
-        # __doc__ (as of 2008-08-02) for pygame.joystick.get_count:
+    @unittest.skipIf(0 == JOYSTICK_COUNT, "No joysticks detected")
```

Comment:
I think it is best to skip this test in a different way.

There is a 'tag' you can apply to a test case class that means all the tests in the class are skipped unless you run the tests with the interactive flag. This is the correct approach to take here as well.

Check out this example in the font_test module:

https://github.com/pygame/pygame/blob/599a76c0e8edf6059001b22fd9e7160d8e050ce8/test/font_test.py#L476-L478

I think we need to basically add a new class to the joystick_test module called `JoystickInteractiveTest` that has the 'interactive' tag as above. That way you can include a basic functionality unit test like your first draft that runs on the CI platforms (Travis, Appveyor) in the `JoystickModuleTest` class, and then a second test in the JoystickInteractiveTest class that does the whole prompt, question scenario in this version of the test. Make sense? Two tests, two different classes. I suspect when other people write more joystick tests they will also fit into this pattern. 

This will also have the advantage of not skipping the interactive test if you start it with no controllers connected.